src/backend/distributed/utils/statistics_collection.c

```diff
@@ -67,36 +71,53 @@ WarnIfSyncDNS(void)
 bool
 CollectBasicUsageStatistics(void)
 {
-	List *distributedTables = NIL;
+	List *distTableOids = NIL;
 	uint64 roundedDistTableCount = 0;
 	uint64 roundedClusterSize = 0;
 	uint32 workerNodeCount = 0;
 	StringInfo fields = makeStringInfo();
 	Datum metadataJsonbDatum = 0;
 	char *metadataJsonbStr = NULL;
+	MemoryContext savedContext = CurrentMemoryContext;
 	struct utsname unameData;
+	bool metadataCollectionFailed = false;
 	memset(&unameData, 0, sizeof(unameData));
 
-	StartTransactionCommand();
+	PG_TRY();
+	{
+		distTableOids = DistTableOidList();
+		roundedDistTableCount = NextPow2(list_length(distTableOids));
+		roundedClusterSize = NextPow2(DistributedTablesSize(distTableOids));
+		workerNodeCount = ActivePrimaryNodeCount();
+		metadataJsonbDatum = DistNodeMetadata();
+		metadataJsonbStr = DatumGetCString(DirectFunctionCall1(jsonb_out,
+															   metadataJsonbDatum));
+	}
+	PG_CATCH();
+	{
+		ErrorData *edata = NULL;
+		MemoryContextSwitchTo(savedContext);
+		edata = CopyErrorData();
+		FlushErrorState();
+
+		/* rethrow as WARNING */
+		edata->elevel = WARNING;
+		ThrowErrorData(edata);
+		metadataCollectionFailed = true;
+	}
+	PG_END_TRY();
 
 	/*
-	 * If there is a version mismatch between loaded version and available
-	 * version, metadata functions will fail. We return early to avoid crashing.
-	 * This can happen when updating the Citus extension.
+	 * Returning here instead of in PG_CATCH() since PG_END_TRY() resets couple
+	 * of global variables.
 	 */
-	if (!CheckCitusVersion(LOG_SERVER_ONLY))
+	if (metadataCollectionFailed)
 	{
-		CommitTransactionCommand();
 		return false;
 	}
-	distributedTables = DistributedTableList();
-	roundedDistTableCount = NextPow2(list_length(distributedTables));
-	roundedClusterSize = NextPow2(ClusterSize(distributedTables));
-	workerNodeCount = ActivePrimaryNodeCount();
-	metadataJsonbDatum = DistNodeMetadata();
-	metadataJsonbStr = DatumGetCString(DirectFunctionCall1(jsonb_out,
-														   metadataJsonbDatum));
-	uname(&unameData);
+
+	/* uname() always succeed if we pass valid buffer */
+	Assert(uname(&unameData) == 0);
```

Comment:
Uhhhh… _please_ do not perform meaningful operations in the body of an `Assert`! The standard PostgreSQL build has asserts disabled; this becomes a no-op!